library/src/scala/tasty/Reflection.scala

```diff
@@ -822,6 +822,15 @@ trait Reflection { reflection =>
 
     /** Matches a block `{ <statements: List[Statement]>; <expr: Term> }` */
     def unapply(x: Block): Option[(List[Statement], Term)]
+
+    /** Creates a block `{ val <name> = <rhs: Term>; <body(x): Term> }` */
+    def let(name: String, rhs: Term)(body: Ident => Term): Term
+
+    /** Creates a block `{ val x = <rhs: Term>; <body(x): Term> }` */
+    def let(rhs: Term)(body: Ident => Term): Term = let("x", rhs)(body)
+
+    /** Creates a block `{ val x1 = <terms(0): Term>; ...; val xn = <terms(n-1): Term>; <body(List(x1, ..., xn)): Term> }` */
+    def let(terms: List[Term])(body: List[Ident] => Term): Term
```

Comment:
I find it more friendly and natural to have `let` out of `Block`. `let` is relatively frequently used, might be better to make it more accessible.

Reply:
Nothing should be out of a module. This method has not been used frequently, only scalatest uses it. If someone wants to use it frequently, they can import it.

Reply:
If everything should be in a module, in my opinion `ValDef` might be a better place to put such methods. But `let` is such a common pattern in (meta-)programming, keeping it at the top-level seems to be the best.

Reply:
Ok. I will move it.

Reply:
Now it is in `ValDef`